bin/sozo/src/ops/account.rs

```diff
@@ -0,0 +1,421 @@
+
+                let simulation_json = colored_json::to_colored_json(
+                    &simulation_json,
+                    ColorMode::Auto(Output::StdOut),
+                )?;
+                println!("{simulation_json}");
+                return Ok(());
+            }
+
+            let account_deployment_tx = account_deployment.send().await?.transaction_hash;
+            eprintln!(
+                "Account deployment transaction: {}",
+                format!("{:#064x}", account_deployment_tx).bright_yellow()
+            );
+
+            // By default we wait for the tx to confirm so that we don't incorrectly mark the
+            // account as deployed
+            eprintln!(
+                "Waiting for transaction {} to confirm. If this process is interrupted, you will \
+                 need to run `{}` to update the account file.",
+                format!("{:#064x}", account_deployment_tx).bright_yellow(),
+                "sozo account fetch".bright_yellow(),
+            );
+            watch_tx(&provider, account_deployment_tx, Duration::from_millis(poll_interval))
+                .await?;
+
+            account.deployment = DeploymentStatus::Deployed(DeployedStatus {
+                class_hash: undeployed_status.class_hash,
+                address: target_deployment_address,
+            });
+
+            // Never write directly to the original file to avoid data loss
+            let mut temp_file_name = file
+                .file_name()
+                .ok_or_else(|| anyhow::anyhow!("unable to determine file name"))?
+                .to_owned();
+            temp_file_name.push(".tmp");
+            let mut temp_path = file.clone();
+            temp_path.set_file_name(temp_file_name);
+
+            let mut temp_file = std::fs::File::create(&temp_path)?;
+            serde_json::to_writer_pretty(&mut temp_file, &account)?;
+            temp_file.write_all(b"\n")?;
+            std::fs::rename(temp_path, file)?;
+        }
+    }
+    Ok(())
+}
+
+fn true_as_default() -> bool {
+    true
+}
+
+fn map_starknet_error(err: StarknetError) -> anyhow::Error {
+    match err {
+        StarknetError::ContractError(err) => {
+            anyhow::anyhow!("ContractError: {}", err.revert_error.trim())
+        }
+        StarknetError::TransactionExecutionError(err) => {
+            anyhow::anyhow!(
+                "TransactionExecutionError (tx index {}): {}",
+                err.transaction_index,
+                err.execution_error.trim()
+            )
+        }
+        StarknetError::ValidationFailure(err) => {
+            anyhow::anyhow!("ValidationFailure: {}", err.trim())
+        }
+        StarknetError::UnexpectedError(err) => {
+            anyhow::anyhow!("UnexpectedError: {}", err.trim())
+        }
+        err => anyhow::anyhow!("{}", err),
+    }
+}
+
+pub async fn watch_tx<P>(
+    provider: P,
+    transaction_hash: FieldElement,
+    poll_interval: Duration,
+) -> Result<()>
```

Comment:
Yes, let's use our waiter instead. At this point it's worth moving out `TransactionWaiter` into a more appropriate crate, so that it can be easily imported by other internal crates and external (Ref https://github.com/cartridge-gg/cairo-webauthn/pull/43). I've done the work in the [`dojo-utils`](https://github.com/dojoengine/dojo/blob/dojo-utils-crate/crates/dojo-utils/src/lib.rs) branch, it seems like a good idea to separate this out.

Reply:
[Add dojo-utils crate](https://github.com/dojoengine/dojo/pull/1601/commits/43a1f6e9e9f793a9f9f9170c03ed60c9125babd4) + [Use TransactionWaiter](https://github.com/dojoengine/dojo/pull/1601/commits/fc1cd0f5f7c9325796939728ce2f37d42bff8a5c)